msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/DefaultRetryPolicy.java

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+/**
+ * Default retry policy for most MSAL Java flows
+ */
+class DefaultRetryPolicy implements IRetryPolicy {
+    private static final int RETRY_NUM = 1;
+    private static int RETRY_DELAY_MS = 1000;
+
+    @Override
+    public boolean isRetryable(IHttpResponse httpResponse) {
+        return httpResponse.statusCode() >= 500 &&
+                httpResponse.statusCode() < 600 &&
+                HttpHelper.getRetryAfterHeader(httpResponse) == null;
+    }
+
+    @Override
+    public int getMaxRetryCount(IHttpResponse httpResponse) {
+        return RETRY_NUM;
+    }
+
+    @Override
+    public int getRetryDelayMs(IHttpResponse httpResponse) {
+        return RETRY_DELAY_MS;
+    }
+
+    //Package-private methods to allow much quicker testing. The delay values should be treated as constants in any non-test scenario.
+    static void setRetryDelayMs(int retryDelayMs) {
+        RETRY_DELAY_MS = retryDelayMs;
+    }
+
+    static void resetToDefaults() {
+        RETRY_DELAY_MS = 1000;
+    }
```

Comment:
I think you should consider setting up a retry policy factory like in .net: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/5308

Additionally, set up a test retry policy factory.

This would eliminate the need for static variables.

What are your thoughts?

Reply:
Should you open a new issue on GitHub to track the refactor? (genuinely curious)